File: android/Ninehoto.Android/Models/AssetGroup.cs
using System.Collections.Generic;
using System.Linq;

namespace Ninehoto.Android.Models
{
    public enum GroupType
    {
        Burst,
        Duplicate,
        Single
    }

    public class AssetGroup
    {
        public string Id { get; }
        public List<MediaItem> Assets { get; }
        public GroupType Type { get; }
        public int Count => Assets.Count;
        public bool IsMultiple => Assets.Count > 1;
        public MediaItem BestAsset => Assets.First();
        public List<long> AssetIds => Assets.Select(a => a.Id).ToList();
        public string FormattedLabel => Type switch
        {
            GroupType.Burst => $"Burst · {Count}",
            GroupType.Duplicate => $"{Count} duplicates",
            _ => ""
        };

        public AssetGroup(List<MediaItem> assets, GroupType type)
        {
            var firstId = assets.FirstOrDefault()?.Id.ToString() ?? Guid.NewGuid().ToString();
            Id = $"{type}_{firstId}";
            Assets = assets;
            Type = type;
        }
    }
}

File: ios/Ninehoto/Models/AssetGroup.swift
import Foundation
import Photos

enum GroupType: String {
    case burst
    case duplicate
    case single
}

struct AssetGroup: Identifiable {
    let id: String
    let assets: [PHAsset]
    let type: GroupType

    var count: Int { assets.count }
    var isMultiple: Bool { assets.count > 1 }
    var bestAsset: PHAsset { assets.first ?? assets[0] }
    var assetIds: [String] { assets.map(\.localIdentifier) }

    var formattedLabel: String {
        switch type {
        case .burst:
            return "Burst · \(count)"
        case .duplicate:
            return "\(count) duplicates"
        case .single:
            return ""
        }
    }

    init(assets: [PHAsset], type: GroupType) {
        let firstId = assets.first?.localIdentifier ?? UUID().uuidString
        self.id = "\(type.rawValue)_\(firstId)"
        self.assets = assets
        self.type = type
    }
}
